src/App.tsx

```diff
@@ -15,36 +18,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const getPreparedGoods = (
+  listOfgoods: string[],
+  sortField: SortField | '',
+  isReversed: boolean,
+): string[] => {
+  const preparedGoods = [...listOfgoods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortField.ALPHABET:
+          return a.localeCompare(b);
+        case SortField.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
```

Comment:
Better would be to create `Default` option in `SortField` enum than to do `SortField | ''`